### PR TITLE
Retains mods except annotations

### PR DESCRIPTION
**Pull request by tabdulradi**
Resolves https://github.com/scalameta/paradise/issues/8

Ready for review

### CODE REVIEW

File: plugin/src/main/scala/org/scalamacros/paradise/typechecker/Expanders.scala [hunk 1]
```diff
@@ -111,7 +111,13 @@ trait Expanders {
...
               case defn: scala.meta.Defn.Macro => defn.copy(mods = Nil)
               case defn: scala.meta.Defn.Type => defn.copy(mods = Nil)
-              case defn: scala.meta.Defn.Class => defn.copy(mods = Nil)
+              case defn: scala.meta.Defn.Class =>
+                val filtredMods = defn.mods.filter {
+                  case scala.meta.Mod.Annot(body: scala.meta.Term) =>
+                    false // TODO: Filter out only the current annotation
+                  case _ => true
+                }
+                defn.copy(mods = filtredMods)
```
Comment (xeno-by) [inline plugin/src/main/scala/org/scalamacros/paradise/typechecker/Expanders.scala]: How about moving the logic of mod filtering in a helper function and applying it consistently to all kinds of decls/defns? We can leave the TODO for future work, but this seems like a very easy win.

Comment (tabdulradi) [inline plugin/src/main/scala/org/scalamacros/paradise/typechecker/Expanders.scala]: Done.

**Comment by xeno-by** [comment]
Thanks for addressing the feedback! The final step is signing the Scala CLA (https://www.lightbend.com/contribute/cla/scala), and then I can merge.

**Comment by tabdulradi** [comment]
Done

**Comment by xeno-by** [comment]
Thanks for your patience! I'll be publishing 3.0.0-SNAPSHOT momentarily.

**Comment by tabdulradi** [comment]
Thanks a lot.

**Pull request merged**